src/shared/components/person/reports.tsx

```diff
@@ -606,40 +722,110 @@ export class Reports extends Component<ReportsRouteProps, ReportsState> {
   async handleResolveCommentReport(form: ResolveCommentReport) {
     const res = await HttpService.client.resolveCommentReport(form);
     this.findAndUpdateCommentReport(res);
-    if (this.state.unreadOrAll === UnreadOrAll.Unread) {
-      this.refetch();
-      UnreadCounterService.Instance.updateReports();
-    }
+    this.update();
   }
 
   async handleResolvePostReport(form: ResolvePostReport) {
     const res = await HttpService.client.resolvePostReport(form);
     this.findAndUpdatePostReport(res);
-    if (this.state.unreadOrAll === UnreadOrAll.Unread) {
-      this.refetch();
-      UnreadCounterService.Instance.updateReports();
-    }
+    this.update();
+  }
+
+  handleRemovePost(form: RemovePost) {
+    this.setState({ removePostForm: form });
```

Comment:
Then these should not be in the state, but call the `client.removePost`

Check out `post.tsx` for how the top level screens should handle these mod actions.